sdv/single_table/base.py

```diff
@@ -604,10 +604,11 @@ def _sample_rows(self, num_rows, conditions=None, transformed_conditions=None,
                 * int:
                     Number of rows that are considered valid.
         """
-        if not self._random_state_set:
-            self._set_random_state(FIXED_RNG_SEED)
+        if (self._model and (
+                self._data_processor.get_sdtypes(primary_keys=False) or keep_extra_columns)):
+            if not self._random_state_set:
+                self._set_random_state(FIXED_RNG_SEED)
```

Comment:
why can't we do this all the time anymore? Also can we add testing for this case?

Reply:
If we don't have a `_model` then this errors (the default implementation calls the `set_random_state` method on `self._model`.